Data/Graph/Inductive/Monad.hs

```diff
@@ -39,7 +44,11 @@ import Data.Graph.Inductive.Graph
 
 -- Monadic Graph
 --
+#if MIN_VERSION_base(4,12,0)
```

Comment:
One way to make this a _little_ less intrusive is to have something like:

```haskell
class
#if MIN_VERSION_base(4,12,0)
  (MonadFail m)
#else
  (Monad m)
#endif
  => GraphM m gr where
```

The advantage of this approach is that you can ensure that `GraphM` is defined regardless in the same way, just that the constraint differs.